src/components/options/providers/main.svelte

```diff
@@ -0,0 +1,44 @@
+<script>
+import AddSearchProviders from "./add.svelte";
+import ContextMenuItems from "./contextMenuItems.svelte";
+import Groups from "./groups.svelte";
+import { onMount } from "svelte";
+
+let groupComponent;
+let contextMenuComponent;
+
+function mainConfigurationUpdated() {
+  updateForms();
+  chrome.runtime.sendMessage({ action: "updateContextualMenu" });
```

Comment:
I just noticed that now we have **two** `mainConfigurationUpdated` functions: one in `options/main.svelte` and another in `options/providers/main.svelte`, which is redundant.

Here you should just emit an event so the function from `options/main.svelte` is executed (i.e. here you do `dispatch("updateMainConfiguration")` and on the parent component you do `<Providers bind:this="{providersTab}" initData="{initData}" on:updateMainConfiguration="{mainConfigurationUpdated}" />`